bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteFlag.java

```diff
@@ -0,0 +1,65 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.client.api;
+
+import com.google.common.base.Preconditions;
+import java.util.EnumSet;
+
+/**
+ * Flags to specify the behaviour of writes
+ */
+public enum WriteFlag {
+    DEFERRED_FORCE(1);
+
+    private final int value;
+
+    WriteFlag(int value) {
+        this.value = value;
+    }
+
+    /**
+     * Converts a set of flags from a binary representation
+     *
+     * @param flagValue the binary value
+     * @return a set of flags
+     */
+    public static EnumSet<WriteFlag> getWriteFlags(int flagValue) {
+        Preconditions.checkArgument(flagValue >= 0 && flagValue <= 1);
```

Comment:
- I think we have a checkstyle rule to use `checkArgument` not `Preconditions.checkArgument`. I am not sure why you can pass checkstyle checking. it is worth to check if checkstyle is enforced in this package or not.

- I am not sure why do you need to verify the flag value here. is it needed?

Reply:
Yes I think it is needed, in the future we do not want Bookies to accept unknown flags.
No we have only flags = 0 or flags = 1, so the check is simpler.

I will update the call to checkArgument and verify checkstyle config

Reply:
@eolivelli I am not just thinking from extensibility. this might potentially limit how old clients interact with new bookies. It is probably fine with WriteFlag, because it is writer initiative. but still it is good to not have such limitation. so I would prefer removing this check.